src/main/java/org/frankframework/http/HttpSenderCached.java

```diff
@@ -0,0 +1,631 @@
+					}
+				}
+			}
+		}
+		return entity.build();
+	}
+
+	protected FormBodyPart elementToFormBodyPart(Element element, PipeLineSession session) throws IOException {
+		String part = element.getAttribute("name"); //Name of the part
+		boolean isFile = "file".equals(element.getAttribute("type")); //text of file, empty == text
+		String filename = element.getAttribute("filename"); //if type == file, the filename
+		String partSessionKey = element.getAttribute("sessionKey"); //SessionKey to retrieve data from
+		String partMimeType = element.getAttribute("mimeType"); //MimeType of the part
+		Message partObject = session.getMessage(partSessionKey);
+		MimeType mimeType = null;
+		if(StringUtils.isNotEmpty(partMimeType)) {
+			mimeType = MimeType.valueOf(partMimeType);
+		}
+
+		final String filenameToUse;
+		if(isFile || StringUtils.isNotBlank(filename)) {
+			String filenamebackup = StringUtils.isBlank(part) ? partSessionKey : part;
+			filenameToUse = StringUtils.isNotBlank(filename) ? filename : filenamebackup;
+		} else {
+			filenameToUse = null;
+		}
+
+		String partname = isFile || StringUtils.isBlank(part) ? partSessionKey : part;
+		return FormBodyPartBuilder.create(partname, new MessageContentBody(partObject, mimeType, filenameToUse)).build();
+	}
+
+	@Override
+	protected Message extractResult(URI targetURI, HttpResponseHandler responseHandler, PipeLineSession session) throws SenderException, IOException {
+		int statusCode = responseHandler.getStatusLine().getStatusCode();
+
+		if (!validateResponseCode(statusCode)) {
+			Message responseBody = responseHandler.getResponseMessage();
+			String body = "";
+			if(responseBody != null) {
+				responseBody.preserve();
+				try {
+					body = responseBody.asString();
+				} catch(IOException e) {
+					body = "(" + ClassUtils.nameOf(e) + "): " + e.getMessage();
+				}
+			}
+			log.warn(getLogPrefix() + "httpstatus [" + statusCode + "] reason [" + responseHandler.getStatusLine().getReasonPhrase() + "]");
+			return new Message(body);
+		}
+
+		Message responseMessage = responseHandler.getResponseMessage();
+		if (!Message.isEmpty(responseMessage)) {
+			responseMessage.closeOnCloseOf(session, this);
+		}
+
+		if (responseHandler.isMultipart()) {
+			return handleMultipartResponse(responseHandler, session);
+		} else {
+			return getResponseBody(targetURI, responseHandler, statusCode);
+		}
+	}
+
+	public Message getResponseBody(URI targetURI, HttpResponseHandler responseHandler, int statusCode) throws IOException {
+		Header[] headers = responseHandler.getAllHeaders();
+		if (getHttpMethod() == HttpMethod.HEAD) {
+			XmlBuilder headersXml = new XmlBuilder("headers");
+			for (Header header : headers) {
+				XmlBuilder headerXml = new XmlBuilder("header");
+				headerXml.addAttribute("name", header.getName());
+				headerXml.setCdataValue(header.getValue());
+				headersXml.addSubElement(headerXml);
+			}
+			return Message.asMessage(headersXml.toXML());
+		}
+
+		super.setCache();
```

Comment:
Can this not be done in the 'configure' method, so that it is executed only once when initializing the sender?